analyzer/analyzer_test.go

```diff
@@ -14,31 +15,29 @@ import (
 func TestAnalyzer(t *testing.T) {
 	t.Parallel()
 	a := analyzer.New()
-	err := a.Flags.Set("err-error", "true")
-	if err != nil {
-		t.Fatalf("failed to set err-error flag")
-	}
-	analysistest.RunWithSuggestedFixes(t, analysistest.TestData(), a, "p")
-}
-
-func TestAnalyzerNoErrError(t *testing.T) {
-	t.Parallel()
-	a := analyzer.New()
-	err := a.Flags.Set("err-error", "false")
-	if err != nil {
-		t.Fatalf("failed to set err-error flag")
-	}
-	analysistest.RunWithSuggestedFixes(t, analysistest.TestData(), a, "nilerror")
-}
+	analysistest.RunWithSuggestedFixes(t, analysistest.TestData(), a, "default")
+	a.Flags.VisitAll(func(f *flag.Flag) {
+		if f.Name != "fiximports" {
+			t.Run(f.Name, func(t *testing.T) {
+				changedVal := "false"
+				if f.DefValue == "false" {
+					changedVal = "true"
+				} else if f.DefValue != "true" {
+					t.Fatalf("default value neither false or true")
+				}
+				err := a.Flags.Set(f.Name, changedVal)
+				if err != nil {
+					t.Fatalf("failed to set err-error flag")
+				}
+				analysistest.RunWithSuggestedFixes(t, analysistest.TestData(), a, f.Name)
+				err = a.Flags.Set(f.Name, f.DefValue)
+				if err != nil {
+					t.Fatalf("failed to set err-error flag")
```

Comment:
Unless I'm wrong TestAnalyzer was only checking err.Error, and you refactored into something able to test everything. 

Thanks for this, it's great. But then it appears to me there is something you left behind


```suggestion
				err := a.Flags.Set(f.Name, changedVal)
				if err != nil {
					t.Fatalf("failed to set %q flag", f.Name)
				}
				analysistest.RunWithSuggestedFixes(t, analysistest.TestData(), a, f.Name)
				err = a.Flags.Set(f.Name, f.DefValue)
				if err != nil {
					t.Fatalf("failed to set %q flag", f.Name)
```

Reply:
Ah yes, thanks